x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -28,23 +28,43 @@
 import org.elasticsearch.xpack.core.security.authc.support.Hasher;
 import org.elasticsearch.xpack.security.support.SecurityIndexManager;
 
+import javax.crypto.SecretKeyFactory;
 import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
 import java.time.Clock;
 import java.time.Instant;
 import java.util.Arrays;
+import java.util.Locale;
+import java.util.function.Function;
 
 import static org.elasticsearch.xpack.core.ClientHelper.SECURITY_ORIGIN;
 import static org.elasticsearch.xpack.core.ClientHelper.executeAsyncWithOrigin;
 
 public class ApiKeyService {
 
-    private static final Logger LOGGER = LogManager.getLogger(ApiKeyService.class);
+    private static final Logger logger = LogManager.getLogger(ApiKeyService.class);
     private static final String TYPE = "doc";
+    public static final Setting<String> PASSWORD_HASHING_ALGORITHM = new Setting<>(
+        "xpack.security.authc.api_key.hashing.algorithm", "pbkdf2", Function.identity(), (v, s) -> {
```

Comment:
nit but maybe `xpack.security.authc.api_key_hashing.algorithm` makes more sense ( consistent with `xpack.security.authc.password_hashing.algorithm` ) ? Naming is hard and I don't have strong opinions

Reply:
Sure I'll be consistent